cartographer/mapping/internal/3d/pose_graph_3d.cc

```diff
@@ -371,6 +371,7 @@ WorkItem::Result PoseGraph3D::ComputeConstraintsForNode(
     }
   }
   constraint_builder_.NotifyEndOfNode();
+  absl::MutexLock locker(&mutex_);
```

Comment:
Doesn't look like you strictly need the mutex for num_nodes_since_last_loop_closure_ since it's only accessed from the work queue, so you could also remove the annotation instead. But this seems fine too, as not much happens here after taking the mutex and until giving it up.